### ISSUE TITLE
claxon needs saxon upgrade?

### ISSUE THREAD

**Issue by paregorios** [paregorios/epidoc_stylesheets_tests#1]
I'm seeing the following error:

```
> ./tests/test_iospe.sh 
Error at xsl:apply-templates on line 64 column 91 of teihi.xsl:
  XTTE0520: Required item type of @select attribute of xsl:apply-templates is node();
  supplied value has item type xs:string
Exception in thread "main" javax.xml.transform.TransformerConfigurationException: Failed to compile stylesheet. 1 error detected.
	at net.sf.saxon.PreparedStylesheet.prepare(PreparedStylesheet.java:249)
	at net.sf.saxon.PreparedStylesheet.compile(PreparedStylesheet.java:114)
	at claxon.claxon$init_xslt$fn__31.invoke(claxon.clj:48)
	at clojure.lang.AFn.call(AFn.java:18)
	at clojure.lang.LockingTransaction.run(LockingTransaction.java:263)
	at clojure.lang.LockingTransaction.runInTransaction(LockingTransaction.java:231)
	at claxon.claxon$init_xslt.invoke(claxon.clj:48)
	at claxon.claxon$_main.doInvoke(claxon.clj:92)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at claxon.claxon.main(Unknown Source)
```

The test script uses our old ```claxon-1.0.0-SNAPSHOT-standalone.jar``` as bundled in the ```bin``` directory and a local clone of the latest head of the EpiDoc Example Stylesheets.

I've tried running the same transformation, with the same version of the EpiDoc Stylesheets and parameters with Oxygen-bundled Saxon-PE 9.8.0.12 and command-line Saxon-HE 9.9.1.7J (installed via HomeBrew). 

**They both run without errors.** E.g.:

```bash
> saxon -s:tests/data/xml/iospe/iospe_1_5.xml -xsl:data/epidoc-stylesheets/start-edition.xsl -o:tests/output/iospe/iospe_1_5.xhtml leiden-style=iospe edn-structure=iospe internal-app-style=iospe external-app-style=iospe
> 
```

This leads me to believe that claxon bundles a down version of saxon that doesn't handle some aspect of our modern stylesheets version.

**Comment by hcayless** [comment]
It probably is an issue with the older version of Saxon. The API has changed significantly, so I'll need to do a spot of rewriting...

**Comment by paregorios** [comment]
Fixed with c13e491
